### PR TITLE
Register Page and Email Verification

### PR DESCRIPTION
**Pull request by mpark4656**
- Create register page
- Implement email verification
- Implement auth trpc router
- Add nodemailer
- Update env to include SMTP data
- Fix next build errors from last pull request

### CODE REVIEW

File: src/components/verify-email/VerifyEmailSuccess.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Link from "next/link";
+
+const VerifyEmailSuccess = () => {
+  return (
+    <section className="min-h-screen flex flex-col justify-center items-center gap-3">
+        Your account has been successfully verified.
+        <Link href="/login" className="text-sm px-3 py-1 rounded border border-[#0A2A75] text-[#0A2A75] hover:bg-[#0A2A75] hover:bg-[#0A2A75] hover:text-white hover:shadow-xl">
```
Comment (ialej001) [inline src/components/verify-email/VerifyEmailSuccess.tsx:7]: There's two `hover:bg` here.

File: src/pages/verify-email/[token].tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const verifyEmailMutation = api.auth.verifyEmail.useMutation();
+
+  useEffect(() => {
+    if(token) {
+      verifyEmailMutation.mutate({ token: String(token) });
+    }
+  }, [token, router]);
+
+  if(verifyEmailMutation.isError) return <VerifyEmailError />
+  if(verifyEmailMutation.isSuccess) return <VerifyEmailSuccess />
```
Comment (ialej001) [inline src/pages/verify-email/[token].tsx:19]: I think it's better to have `VerifyEmailSuccess` as the default returned component. Basically swap this with the loading fragment and change the condition to `verifyEmailMutation.isLoading`

**Review by ialej001 — APPROVED** [review]
Solid work Michael.

File: src/components/verify-email/VerifyEmailSuccess.tsx (hunk 1 above)
Comment (mpark4656) [reply src/components/verify-email/VerifyEmailSuccess.tsx:7]: Thank you. Fixed

File: src/pages/verify-email/[token].tsx (hunk 2 above)
Comment (mpark4656) [reply src/pages/verify-email/[token].tsx:19]: I tried your method but found a minor issue. I see success component initially flashing when visiting the page and then goes to error (This is assuming the verify URL was bad or token was invalid).

I think initially while the component is mounting, the state of verifyEmailMutation is neither isError, isSuccess, or isLoading, so the default return of success component is shown briefly until the state changes to isLoading.

Comment (ialej001) [reply src/pages/verify-email/[token].tsx:19]: Ah, I see. If it works as it is without use an `useEffect()` hook, it's probably for the best.

**Review by Briantam0422 — APPROVED** [review]
Nice! Thank you for fixing the build error!!!

**Pull request merged**